src/pycolmap/estimators/covariance.cc

```diff
@@ -0,0 +1,198 @@
+#include "colmap/estimators/covariance.h"
+
+#include "pycolmap/pybind11_extension.h"
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+using namespace colmap;
+using namespace pybind11::literals;
+namespace py = pybind11;
+
+namespace {
+
+void ConvertListOfPyArraysToConstPointers(
+    const std::vector<py::array_t<double>>& pyarrays,
+    std::vector<const double*>& blocks) {
```

Comment:
You could create `blocks` in this function instead of forcing the parent scope to do so.

Reply:
How do you achieve that without copying the blocks?

Reply:
Return value optimization - there shouldn't be any copy if default optimizations are enabled.